android/src/main/java/com/appgoalz/rnjwplayer/RNJWPlayerView.java

```diff
@@ -290,62 +302,119 @@ public void setupPlayerView() {
             mPlayer.addOnControlBarVisibilityListener(this);
             mPlayer.addOnDisplayClickListener(this);
             mPlayer.addOnFullscreenListener(this);
-            mPlayer.setFullscreenHandler(new FullscreenHandler() {
-                @Override
-                public void onFullscreenRequested() {
-                    WritableMap eventEnterFullscreen = Arguments.createMap();
-                    eventEnterFullscreen.putString("message", "onFullscreen");
-                    getReactContext().getJSModule(RCTEventEmitter.class).receiveEvent(
-                            getId(),
-                            "topFullScreen",
-                            eventEnterFullscreen);
-                }
+            mPlayer.setFullscreenHandler(new AppViewFullscreenHandler(mPlayer));
+            mPlayer.setControls(true);
+            mPlayer.setBackgroundAudio(true); // TODO: - add as prop
+        }
+    }
 
-                @Override
-                public void onFullscreenExitRequested() {
-                    WritableMap eventExitFullscreen = Arguments.createMap();
-                    eventExitFullscreen.putString("message", "onFullscreenExit");
-                    getReactContext().getJSModule(RCTEventEmitter.class).receiveEvent(
-                            getId(),
-                            "topFullScreenExit",
-                            eventExitFullscreen);
-                }
+    private class AppViewFullscreenHandler implements FullscreenHandler {
+        private ViewGroup mPlayerContainer;
+        private final RNJWPlayer mPlayer;
+        private View mDecorView;
 
-                @Override
-                public void onResume() {
+        public AppViewFullscreenHandler(RNJWPlayer player) {
+            mPlayerContainer = (ViewGroup) player.getParent()
+            mPlayer = player;
+        }
 
-                }
+        @Override
+        public void onFullscreenRequested() {
+            mDecorView = mActivity.getWindow().getDecorView();
 
-                @Override
-                public void onPause() {
+            // Hide system ui
+            mDecorView.setSystemUiVisibility(
+                View.SYSTEM_UI_FLAG_HIDE_NAVIGATION // hides bottom bar
+                | View.SYSTEM_UI_FLAG_FULLSCREEN // hides top bar
+                | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY // prevents navigation bar from overriding
+                // exit-full-screen button. Swipe from side to access nav bar.
+            );
 
-                }
+            // Enter landscape mode for fullscreen videos
+            mActivity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
```

Comment:
ActivityInfo is missing symbol... There is no ActivityInfo

Reply:
Yep, thanks!